homeassistant/components/recorder/models.py

```diff
@@ -619,7 +619,10 @@ def last_changed(self, value: datetime) -> None:
     def last_updated(self) -> datetime:  # type: ignore[override]
         """Last updated datetime."""
         if self._last_updated is None:
-            self._last_updated = process_timestamp(self._row.last_updated)
+            if (last_updated := self._row.last_updated) is not None:
+                self._last_updated = process_timestamp(last_updated)
+            else:
+                self._last_updated = process_timestamp(self._row.last_changed)
```

Comment:
Not sure if this is faster, but it might potentially avoid parsing a timestamp
```suggestion
                self._last_updated = self.last_changed
```

Reply:
Thanks